plugins/org.jkiss.dbeaver.model.sql/src/org/jkiss/dbeaver/model/sql/parser/SQLRuleManager.java

```diff
@@ -89,12 +89,19 @@ public TPRule[] getRulesByType(@NotNull SQLTokenType requiredType) {
     }
 
     public void loadRules() {
-        loadRules(null, false);
+        loadRules((DBPDataSourceContainer) null, false);
     }
 
     public void loadRules(@Nullable DBPDataSource dataSource, boolean minimalRules) {
+        if (dataSource == null) {
+            loadRules((DBPDataSourceContainer) null, minimalRules);
+        } else {
+            loadRules(dataSource.getContainer(), minimalRules);
```

Comment:
Now you get data source container from data source and then in the loadRules method you get data source from that container. Why not passing datasource directly then?

Reply:
it brings a lot of changes